lib/charms/mongodb/v0/shards_interface.py

```diff
@@ -126,6 +130,12 @@ def pass_hook_checks(self, event: EventBase) -> bool:
             event.defer()
             return False
 
+        # adding/removing shards while a backup/restore is in progress can be disastrous
+        pbm_status = self.charm.backups._get_pbm_status()
```

Comment:
Is it possible to avoid using of private methods of backups?  😄 

I see that here we are checking status of charm, right?

Reply:
> Is it possible to avoid using of private methods of backups? 😄

This makes me think. We actually use `_get_pbm_status` within `src/charm.py`, perhaps I should make this function `public`. What do you think?

> I see that here we are checking status of charm, right?

No we should be checking the status of backups

Reply:
> What do you think?
Yeah, I think we can mark it as public if it is used outside backups

Reply:
Thanks for the guidiance, I am on it!